### PR TITLE
Implement multiple login ID per key and login ID key config

### PR DESCRIPTION
**Pull request by kiootic**
refs #933
depends on #947

### CODE REVIEW

File: pkg/auth/dependency/sso/sso.go [hunk 1]
```diff
@@ -97,10 +97,14 @@ type AuthInfo struct {
 	ProviderUserID          string
 	ProviderUserProfile     map[string]interface{}
 	ProviderAccessTokenResp interface{}
-	ProviderAuthData        map[string]string
+	ProviderAuthData        ProviderAuthKeys
 	State                   State
 }
 
+type ProviderAuthKeys struct {
```
Comment (louischan-oursky) [inline pkg/auth/dependency/sso/sso.go]: IMO, `ProviderUserInfo` is a better name.

File: pkg/auth/handler/signup.go [hunk 2]
```diff
@@ -147,6 +148,7 @@ func (h SignupHandler) DecodeRequest(request *http.Request) (handler.RequestPayl
 	if payload.Realm == "" {
 		payload.Realm = password.DefaultRealm
 	}
+	payload.LoginIDs = password.ParseLoginIDs(payload.RawLoginIDs)
```
[the remark this answers is not in the document]
Comment (kiootic) [reply pkg/auth/handler/signup.go]: `ParseLoginIDs` only parses the raw list of maps into list of key value pairs; all possible input of `RawLoginIDs` is valid for this function. The validation is performed in the `Validate` function of the payload instead.

File: pkg/auth/dependency/sso/sso.go (hunk 1 above)
Comment (kiootic) [reply pkg/auth/dependency/sso/sso.go]: This struct is intended to store only auth info (e.g. email) for matching with login IDs according to current SSO code. I agree changing it to `ProviderUserInfo` would be better, but then we should refactor the `DecodeXXX` functions in `AuthInfoProcessor` interface into single `DecodeUserInfo` function. We can leave this for later?

Comment (louischan-oursky) [reply pkg/auth/dependency/sso/sso.go]: It is good refactoring! When should we do this refactoring if not now?

File: pkg/auth/handler/signup.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (louischan-oursky) [reply pkg/auth/handler/signup.go]: Let's adjust the HTTP API to accept LoginID directly. The client SDK should flexibly accept different kind of shorthands and convert for the server.

File: pkg/auth/dependency/sso/sso.go (hunk 1 above)
Comment (kiootic) [reply pkg/auth/dependency/sso/sso.go]: Will refactor SSO related code in #957

File: pkg/auth/handler/userverify/verify_request.go [hunk 3]
```diff
@@ -56,16 +59,32 @@ func (f VerifyRequestHandlerFactory) ProvideAuthzPolicy() authz.Policy {
...
+	case loginIDTypeEmail:
+		return provider.CheckLoginIDKeyType(principal.LoginIDKey, metadata.Email)
+	default:
+		return false
 	}
+}
+
+type VerifyRequestPayload struct {
+	RawLoginIDType string      `json:"login_id_type"`
+	LoginIDType    loginIDType `json:"-"`
```
Comment (louischan-oursky) [inline pkg/auth/handler/userverify/verify_request.go]: `RawLoginIDType` seems to redundant. We have `Validate` so we can validate `LoginIDType loginIDType` before `Handle` can ever be run

**Pull request merged**